src/net_processing.cpp

```diff
@@ -321,10 +321,17 @@ struct CNodeState {
      */
     bool fSupportsDesiredCmpctVersion;
 
-    /** State used to enforce CHAIN_SYNC_TIMEOUT
-      * Only in effect for outbound, non-manual, full-relay connections, with
-      * m_protect == false
-      * Algorithm: if a peer's best known block has less work than our tip,
+    /** State used to enforce CHAIN_SYNC_TIMEOUT and EXTRA_PEER_CHECK_INTERVAL logic.
+      *
+      * Both are only in effect for outbound, non-manual, non-protected connections.
+      * Any peer protected (m_protect = true) is not chosen for eviction. A peer is
+      * marked as protected if all of these are true:
+      *   - its connection type is IsBlockOnlyConn() == false
+      *   - it gave us a valid connecting header
+      *   - we haven't reached MAX_OUTBOUND_PEERS_TO_PROTECT_FROM_DISCONNECT yet
+      *   - it has a better chain than we have
```

Comment:
I don't think this is quite right? 

the relevant part of the check in `ProcessHeadersMessage` checks `nodestate->pindexBestKnownBlock->nChainWork >= ::ChainActive().Tip()->nChainWork`, aka would set `m_protect` to true if they have the same chain tip as we do. 

I think saying the peer "has a better chain" is misleading, because then I'd expect us to usually not have protected peers. Based on the logic I see, I'd expect us to usually have `MAX_OUTBOUND_PEERS_TO_PROTECT_FROM_DISCONNECT`.

what do you think?